src/Serializer/AbstractItemNormalizer.php

```diff
@@ -88,7 +88,12 @@ public function __construct(PropertyNameCollectionFactoryInterface $propertyName
         $this->resourceClassResolver = $resourceClassResolver;
         $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
         $this->itemDataProvider = $itemDataProvider;
+
+        if (true === $this->allowPlainIdentifiers) {
+            @trigger_error(sprintf('Allowing plain identifiers as argument of "%s" is deprecated since API Platform 2.7 and will not be possible anymore in API Platform 3.', self::class), \E_USER_DEPRECATED);
```

Comment:
do you think we could try and run our tests without plain identifiers to see what would potentially break ?

Reply:
I tried turning `allow_plain_identifiers` to false and run Behat tests, the only failing scenario are: features/json/relation.feature:154 & 187. I added a `@deprecated` tag and a TODO on them